frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -275,3 +283,77 @@ export function useKubeObjectList<T extends KubeObjectClass>({
     },
   };
 }
+
+/**
+ * Returns a combined list of Kubernetes objects and watches for changes from the clusters given.
+ */
+export function useKubeObjectLists<T extends KubeObjectClass>({
+  kubeObjectClass,
+  namespace,
+  clusters,
+  queryParams,
+}: {
+  /** Class to instantiate the object with */
+  kubeObjectClass: T;
+  /** Object list namespace */
+  namespace?: string;
+  /** Object list clusters */
+  clusters: string[];
+  queryParams?: QueryParameters;
+}): [Array<InstanceType<T>> | null, ApiError | null] &
+  QueryListResponse<KubeList<InstanceType<T>>, InstanceType<T>, ApiError> {
+  const clusterResults = clusters.reduce((acc, cluster) => {
+    acc[cluster] = useKubeObjectList<T>({
+      kubeObjectClass,
+      namespace,
+      cluster,
+      queryParams,
+    });
+    return acc;
+  }, {} as Record<string, ReturnType<typeof useKubeObjectList<T>>>);
+
+  const items = clusters.flatMap(cluster => {
+    const items = clusterResults[cluster].items ?? [];
+    return items.map(item => ({ ...item, cluster }));
+  });
+  // data makes no sense really for multiple clusters, but useful for single cluster?
+  const data =
+    clusters.map(cluster => clusterResults[cluster].data).find(it => it !== null) ?? null;
+  const error =
+    clusters.map(cluster => clusterResults[cluster].error).find(it => it !== null) ?? null;
+  const isError = clusters.some(cluster => clusterResults[cluster].isError);
+  const isLoading = clusters.some(cluster => clusterResults[cluster].isLoading);
+  const isFetching = clusters.some(cluster => clusterResults[cluster].isFetching);
+  const isSuccess = clusters.every(cluster => clusterResults[cluster].isSuccess);
+  // status makes no sense really for multiple clusters, but maybe useful for single cluster?
+  const status =
+    clusters.map(cluster => clusterResults[cluster].status).find(it => it !== null) ?? 'pending';
+
+  let clusterErrors: Record<string, ApiError | null> | null = {};
+  clusters.forEach(cluster => {
+    if (clusterErrors && clusterResults[cluster]?.error !== null) {
+      clusterErrors[cluster] = clusterResults[cluster].error;
+    }
+  });
+  if (Object.keys(clusterErrors).length === 0) {
+    clusterErrors = null;
+  }
+
+  // @ts-ignore
+  return {
+    items,
+    data,
+    error,
+    isError,
+    isLoading,
+    isFetching,
+    isSuccess,
+    status,
+    *[Symbol.iterator](): ArrayIterator<ApiError | InstanceType<T>[] | null> {
+      yield items;
+      yield error;
+    },
```

Comment:
I think this is only for the return to work also as a list. Since there's no backward compat need for this function, we can remove this.

Reply:
plugins may need backward compatibility